src/main/java/uk/gov/pay/directdebit/payments/services/PaymentSearchService.java

```diff
@@ -22,31 +21,26 @@
     private UriInfo uriInfo;
 
     @Inject
-    public PaymentSearchService(PaymentViewDao paymentViewDao, GatewayAccountDao gatewayAccountDao) {
+    PaymentSearchService(PaymentViewDao paymentViewDao, GatewayAccountDao gatewayAccountDao) {
         this.paymentViewDao = paymentViewDao;
         this.gatewayAccountDao = gatewayAccountDao;
     }
 
-    public PaymentViewResponse getPaymentSearchResponse(PaymentViewSearchParams searchParams) {
-        return gatewayAccountDao.findByExternalId(searchParams.getGatewayExternalId())
-                .map(gatewayAccount -> {
-                    PaymentViewSearchParams validatedSearchParams = paymentViewValidator.validateParams(searchParams);
-                    List<PaymentResponse> viewListResponse = Collections.emptyList();
-                    Long total = getTotal(validatedSearchParams);
-                    if (total > 0) {
```

Comment:
Why was this condition removed? Optimisation just not worth it?

Reply:
Going to add it back.